hrjobcontract/CRM/Hrjobcontract/ExportImportValuesConverter.php

```diff
@@ -355,79 +355,77 @@ public function pay_pay_scale_export($value)
         }
         return $result;
     }
-    public function pay_pay_scale_import($value)
-    {
-        if (!isset($value))
-        {
-            return null;
-        }
-        $keys = preg_split("/( - | per )/", $value);
-        $parts = array();
-        if (isset($keys[2]))
-        {
-            $parts = explode(' ', $keys[2]);
-        }
+    
+    /**
+     * Obtains id for given pay scale. If given payscale does not exist in DB,
+     * it will attempt to create it. If given payscale follows the format
+     * "%s - %s %d per %s", the information will be used to populate currency, 
+     * amount and periodicity fields for the pay scale.  Otherwise it will just 
+     * add it with the given label and leave all other values empty.
+     * 
+     * @param string $value
+     *   String for pay scale / grade that needs to be imported
+     * 
+     * @return int
+     *   ID of given pay scale / grade
+     */
+    public function pay_pay_scale_import($value) {
+      if (!isset($value)) {
+        return null;
+      }
+
+      $payScale = trim(substr($value, 0, (strrpos($value, '-'))));
+      $payData = trim(substr($value, (strrpos($value, '-') + 1)));
+      $parts = explode(' ', $payData);
+
+      if (count($parts) === 4 && !empty($payScale)) {
+        $payScaleParams = [
+          'pay_scale' => $payScale,
+          'currency' => $parts[0],
+          'amount' => $parts[1],
+          'periodicity' => $parts[3]
+        ];
+      } else {
+        $payScaleParams = [
+          'pay_scale' => $value
+        ];
+      }
+
+      $result = civicrm_api3('HRPayScale', 'get', array_merge(
+        ['sequential' => 1],
+        $payScaleParams
+      ));
+
+      if ($result['count'] == 1) {
+        return $result['id'];
+      } 
+      elseif ($result['count'] > 1) {
+        $payScaleRecord = array_shift($result['values']);
```

Comment:
you can just use the  do : 

```php
return ($result['count'] > 0)  ?   array_shift($result['values'])['id'] :  $this->createPayScale($payScaleParams);
```
also return null; will not be reached so it could be removed